packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -53,30 +53,48 @@ struct UnresolvedWitnessIndexes {
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait BuildableTransaction: sealed::Sealed {
     type TxType: Transaction;
+    type Context;
 
-    async fn build(self, provider: impl DryRunner) -> Result<Self::TxType>;
+    async fn build(self, provider: impl DryRunner, context: Self::Context) -> Result<Self::TxType>;
+}
+
+impl sealed::Sealed for ScriptTransactionBuilder {}
 
+#[derive(Debug, Clone, Default)]
+pub enum ScriptContext {
+    /// Transaction is estimated and signatures are automatically added.
+    #[default]
+    Normal,
```

Comment:
I just have this tiny nit regarding the naming `Normal`.  I think id does not reflect well what we are doing. What about something like: `CompleteTx`, `FinalizedTx` ... something that would mean that we did all the work in the background.

Reply:
I vote for `Complete` or `Full`, the tx is implied already